app/views/job_templates/_job_template_callback_tab_headers.html.erb

```diff
@@ -0,0 +1,13 @@
+<li><a id="ansible_tab_header" href="#ansible_callback_enabled" data-toggle="tab"><%= _("Ansible") %></a></li>
+<script type="text/javascript">
+    $(document).ready(function () {
+        var provider_type = $('#job_template_provider_type');
+        provider_type.change(ansibleTabVisibilityByProvider);
+        provider_type.change();
+
+        function ansibleTabVisibilityByProvider() {
+            var tab_header = $("#ansible_tab_header");
+            $("#select2-chosen-1").text() === 'Ansible' ?  tab_header.show() : tab_header.hide();
```

Comment:
This function can take an argument - the element that changed. From it we can pull the value without resorting to touching select2's internals.